### ISSUE TITLE
Accessibility of component types for screenreaderes

### ISSUE THREAD

**Issue by livvyfox** [reskit/moodle-tiny_c4l#19]
**Describe the bug**
Not all component types are being read by a screenreader

**To Reproduce**
1. Create a Page resource
2. Add a number of components including - Tip, Attention, Estimated Time, Do/don't cards.
3. Download & install free screenreader - NVDA
4. Read the C4L components using a screenreader - NVDA. The screenreader is not reading the ARIA-Labels for a number of the components. However it does read Estimated time as expected

**Expected behavior**
I expect the screenreader to read out the different component types to ensure that screenreader users are not missing any context for the content they are reviewing e.g. Attention, tip, do or do not.

The Estimated time component is being read out by the screenreader so it seems like this is expected behaviour for all component types.

**Desktop (please complete the following information):**
- Windows
- Chrome 
 - Version 137.0.7151.122
 - Also tested on Firefox in case it was browser related 140.0.1
 - NVDA version 2025.1.2

**Comment by mfullerkillgore** [comment]
I tested this as well and was able to reproduce the same behavior in Firefox using both NVDA 2025.3.3 and Windows Narrator.

In both cases, the screen reader read only the visible text content and did not announce the component type from the aria-label. So for elements like Tip, Attention, etc., the contextual label was not conveyed to the user unless it was visibly present in the content itself.

That seems to match what was reported here, and it suggests that having the aria-label on the current container alone is not sufficient for these components to be announced reliably by screen readers.

Our university was very interested in using this plugin, but with this accessibility concern we will not be able to move forward with it in production. For us, this is an important blocker because those component labels provide meaningful context for screen reader users.

**Comment by mfullerkillgore** [comment]
I have retested this tool with the two most recent versions and although I see where role="note" was added to the components I am testings when I inspect the html the aria-label are still not being read out by NVDA or my Windows Narrator. Tested in both Firefox and Chrome. I'm also testing on a PC.

@livvyfox were you able to test and verify this fix? @rogersegu  or @reskit what screen reader are you using to test with?

**Comment by mfullerkillgore** [comment]
I did some additional testing and found that NVDA will read the aria-label when hovering the element with the mouse, but that does not seem reliable for normal keyboard navigation. A more dependable fix was adding hidden text directly in the DOM, for example <span class="sr-only">Tip: </span>, which worked with both Windows Narrator and NVDA. Based on that testing, using an sr-only text label appears to be more accessible than relying on aria-label alone for these static callouts.
